Source/WebCore/dom/Element.cpp

```diff
@@ -2352,26 +2352,48 @@ static RefPtr<Element> getElementByIdIncludingDisconnected(const Element& startE
     return nullptr;
 }
 
-RefPtr<Element> Element::getElementAttribute(const QualifiedName& attributeName) const
+RefPtr<Element> Element::getElementForAttributeInternal(const QualifiedName& attributeName) const
 {
-    ASSERT(isElementReflectionAttribute(document().settings(), attributeName));
+    RefPtr<Element> element = nullptr;
+    bool hasExplicitlySetElement = false;
 
     if (auto* map = explicitlySetAttrElementsMapIfExists()) {
         auto it = map->find(attributeName);
         if (it != map->end()) {
             ASSERT(it->value.size() == 1);
-            RefPtr element = it->value[0].get();
-            if (element && isDescendantOrShadowDescendantOf(element->rootNode()))
-                return element;
-            return nullptr;
+            hasExplicitlySetElement = true;
+            RefPtr explicitlySetElement = it->value[0].get();
+            if (explicitlySetElement && isDescendantOrShadowDescendantOf(explicitlySetElement->rootNode()))
+                element = explicitlySetElement;
         }
     }
+    
+    if (!hasExplicitlySetElement) {
+    const AtomString& id = getAttribute(attributeName);
+        element = getElementByIdInternalIncludingDisconnected(*this, id);
+    }
+
+    if (element == nullptr)
+        return nullptr;
+
+    return element->deepShadowRootReferenceTargetOrSelf();
+}
+
+RefPtr<Element> Element::getElementAttributeForBindings(const QualifiedName& attributeName) const
+{
+    ASSERT(isElementReflectionAttribute(document().settings(), attributeName));
+    RefPtr<Element> element = getElementForAttributeInternal(attributeName);
```

Comment:
I believe it's more common to not include the `<Element>` type specification when you can do so, i.e. `RefPtr` instead of `RefPtr<Element>`

Reply:
I think for conciseness when the context makes it reasonably clear what type the `RefPtr` is going to hold. But I looked and this isn't officially codified in our style guidelines, so I think it could be fine to leave as-is too.